app/src/androidTest/java/com/github/houseorganizer/houseorganizer/MainActivityTest.java

```diff
@@ -3,44 +3,87 @@
 import static androidx.test.espresso.Espresso.onView;
 import static androidx.test.espresso.assertion.ViewAssertions.matches;
 import static androidx.test.espresso.intent.Intents.intended;
-import static androidx.test.espresso.intent.matcher.IntentMatchers.toPackage;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
 
-import android.content.Intent;
-
-import androidx.test.core.app.ActivityScenario;
-import androidx.test.core.app.ApplicationProvider;
 import androidx.test.espresso.intent.Intents;
 import androidx.test.ext.junit.rules.ActivityScenarioRule;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
 
+import com.google.firebase.auth.FirebaseAuth;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
 
 @RunWith(AndroidJUnit4.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class MainActivityTest {
 
     @Rule
     public ActivityScenarioRule<MainActivity> testRule = new ActivityScenarioRule<>(MainActivity.class);
 
+    /*
+    @BeforeClass
+    public static void initializingDatabase(){
+        FirebaseAuth mAuth = FirebaseAuth.getInstance();
+        if(!FirebaseTestsHelper.emulatorActivated){
+            mAuth.useEmulator("10.0.2.2", 9099);
+            FirebaseTestsHelper.emulatorActivated = true;
+        }
+        mAuth.createUserWithEmailAndPassword(TEST_USER_EMAIL, TEST_USER_PASSWORD);
+        /*if(mAuth.getCurrentUser() != null){
+            mAuth.signOut();
+        }
+        Intents.init();
+    }*/
+
+    /*@Before
+    public void initIntents(){
+        Intents.init();
+    }/*
+
     /* See logo */
     @Test
-    public void seeHouseLogo() {
+    public void test1seeHouseLogo() {
```

Comment:
Perhaps the test names in this file could be changed, but this is a task for another sprint.

Reply:
Yes this whole file needs to be remade as hours of testing couldn't get it to work. The names were a (failed) attempt to order the execution.